The SRT_ATR_ALIGNAS macro redefinintion

The commit 5f57dde37fcaaa79f972162ccc4bba79e6b95996 (@yomnes0) occasionally directly pushed to master introduced this issue.
The `SRT_ATR_ALIGNAS` macro is added before the `#ifndef INC_SRT_CHANNEL_H` check, thus generating the following error:


```shell
[ 10%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/channel.cpp.o
In file included from srtcore/core.h:66,
                 from srtcore/channel.cpp:61:
srtcore/channel.h:53: error: "SRT_ATR_ALIGNAS" redefined [-Werror]
   53 | #define SRT_ATR_ALIGNAS(n) alignas(n)
      |
In file included from srtcore/channel.cpp:60:
srtcore/channel.h:57: note: this is the location of the previous definition
   57 | #define SRT_ATR_ALIGNAS(n)
      |
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/srt_virtual.dir/build.make:146: CMakeFiles/srt_virtual.dir/srtcore/channel.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:95: CMakeFiles/srt_virtual.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```


Those definitions should be placed in the `srt_attr_defs.h`.

--- a/srtcore/channel.h
+++ b/srtcore/channel.h
@@ -49,13 +49,6 @@ written by
 modified by
    Haivision Systems Inc.
 *****************************************************************************/
-#if HAVE_CXX11
-#define SRT_ATR_ALIGNAS(n) alignas(n)
-#elif HAVE_GCC
-#define SRT_ATR_ALIGNAS(n) __attribute__((aligned(n)))
-#else
-#define SRT_ATR_ALIGNAS(n)
-#endif
 #ifndef INC_SRT_CHANNEL_H
 #define INC_SRT_CHANNEL_H
 

--- a/srtcore/srt_attr_defs.h
+++ b/srtcore/srt_attr_defs.h
@@ -31,6 +31,14 @@ used by SRT library internally.
 #define ATR_DEPRECATED
 #endif
 
+#if HAVE_CXX11
+#define SRT_ATR_ALIGNAS(n) alignas(n)
+#elif HAVE_GCC
+#define SRT_ATR_ALIGNAS(n) __attribute__((aligned(n)))
+#else
+#define SRT_ATR_ALIGNAS(n)
+#endif
+
 #if defined(__cplusplus) && __cplusplus > 199711L
 #define HAVE_CXX11 1
 // For gcc 4.7, claim C++11 is supported, as long as experimental C++0x is on,

